src/qibo/tests/test_error_mitigation_cdr.py

```diff
@@ -0,0 +1,84 @@
+import numpy as np
+import pytest
+
+from qibo import gates
+from qibo.hamiltonians import SymbolicHamiltonian
+from qibo.models import Circuit
+from qibo.models.error_mitigation import CDR, sample_training_circuit
+from qibo.noise import DepolarizingError, NoiseModel
+from qibo.symbols import Z
+
+
+def get_noise_model(error):
+    noise = NoiseModel()
+    noise.add(error, gates.CNOT)
+    return noise
+
+
+@pytest.mark.parametrize("nqubits", [3])
+@pytest.mark.parametrize(
+    "noise",
+    [get_noise_model(DepolarizingError(0.1))],
+)
+@pytest.mark.parametrize("full_output", [False, True])
+def test_cdr(nqubits, noise, full_output):
```

Comment:
Could you please add a `backend` argument in all tests that are executing circuits:
```suggestion
def test_cdr(backend, nqubits, noise, full_output):
```
Note that you do not need to define this argument. It is a global fixture so pytest will pass the proper values automatically.